packages/phpstan-rules/packages/nette/src/Latte/Tokens/PhpToLatteLineNumbersResolver.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Symplify\PHPStanRules\Nette\Latte\Tokens;
+
+use Nette\Utils\Strings;
+
+final class PhpToLatteLineNumbersResolver
```

Comment:
@lulco I've noticed your solution uses PHP lines and not original Latte lines, so it's not useful as you asked in Twitter. 
This services solves it, as it maps PHP line numbers back to Latte line numebrs. Feel free to copy it.

Reply:
Hmm it should not. After parsing latte I use lines from tokenizer as StartLine attribute in nodes. I'll check it again

Reply:
Ah, I didn't find that in your code probably. How do you traverse nodes with custom attributes for PHPStan?

Reply:
It is here https://github.com/efabrica-team/phpstan-latte/blob/c3af90a1935b4e6a9ce15a1ef80f0e765ec2aceb/src/Parser/LatteParser.php#L131

Sorry, I probably don’t understand the question :) I am new in custom phpstan rules (actually this is my first). What do you mean with custom atrributes and traversing? For now I'm just creating Nodes based on latte

Reply:
Ah, I see. So you're basically creating PHP nodes manually from Latte token? Intersting approach!

IMO that only duplicated what php-parser does nativelly and will require lot of effort to cover all edge cases.

I handle the same input to output here using Latte\Parser + nikic\php-parser: https://github.com/symplify/symplify/pull/3568/files#diff-6bcf54852a2fc4853be302c7df284c722e39859268a98849ac004e4d2d3a1f79R31-R44

Reply:
> Ah, I see. So you're basically creating PHP nodes manually from Latte token? Intersting approach!

Yes I do.

> IMO that only duplicates what php-parser does nativelly and will require lot of effort to cover all edge cases.

Probably you are right, there are several edge cases, but I wanted to try it at least :D I feel templates are the last thing in our applications which is not tested very well. And it deserve to be tested, there are many potential errors in templates.

> I handle the same input to output here using Latte\Parser + nikic\php-parser: https://github.com/symplify/symplify/pull/3568/files#diff-6bcf54852a2fc4853be302c7df284c722e39859268a98849ac004e4d2d3a1f79R31-R44

I'll check it, thanks

Reply:
> And it deserve to be tested, there are many potential errors in templates.

🙏👍